src/dinitctl.cc

```diff
@@ -484,6 +564,12 @@ int dinitctl_main(int argc, char **argv)
             }
             return cat_service_log(socknum, rbuffer, service_name, catlog_clear);
         }
+        else if (command == command_t::SIG_SEND) {
+            return signal_send(socknum, rbuffer, service_name, sig_num);
+        }
+        else if (command == command_t::SIG_LIST) {
+            return signal_list();
```

Comment:
This doesn't need a control socket connection, but one has been opened anyway. That should really be fixed. I.e. don't open a control socket for a command that doesn't need it.